src/api/db/migrate/20230414085150_add_notified_projects_project_id_index.rb

```diff
@@ -0,0 +1,5 @@
+class AddNotifiedProjectsProjectIdIndex < ActiveRecord::Migration[7.0]
+  def change
+    add_index :notified_projects, :project_id
```

Comment:
Might be something we have to schedule for the maintenance window then? Did you check how long this takes?

```
[2] pry(main)> NotifiedProject.count
=> 4350704
```

Reply:
[Strong migrations](https://github.com/ankane/strong_migrations#safe-by-default) says it's safe to add an index. So it should not lock the table and be safe to do it outside the maintenance window.

Reply:
Hm but we don't use that setting...

Reply:
After reading [MariaBD documentation regarding adding indexes](https://mariadb.com/kb/en/innodb-online-ddl-operations-with-the-inplace-alter-algorithm/#alter-table-add-primary-key) we can ask for a specific locking mechanism to be used, but if that strategy cannot be applied, the migration will fail. If we don't ask for a specific locking mechanism the default one kicks in and then it tries to acquires the least restrictive one at runtime. So I think it's better to schedule this to be run during the maintenance window in case the MariaDB engine chooses to lock the table. WDYT?

Reply:
sounds about right

Reply:
As for how long will it take, it depends on the algorithm the engine uses to perform the alter table... From ` incredibly slow` for the `ALGORITHM=COPY`, to almost instantaneous if `ALGORITHM=INSTANT` is chosen, but I haven't yet found how to tell ActiveRecord to let me choose an alter table algorithm...